test/mobile/op_deps/CMakeLists.txt

```diff
@@ -0,0 +1,48 @@
+cmake_minimum_required(VERSION 3.1)
+
+project(op_deps_project)
+
+# Find torch library
+add_library(torch UNKNOWN IMPORTED)
+find_package(Torch REQUIRED)
+include_directories(${TORCH_INCLUDE_DIRS})
+link_directories(${TORCH_LIBRARIES})
+
+# Small test op library
+add_library(SimpleOps
+  simple_ops.cpp
+  utils.cpp
+)
+
+# Main executable
+add_executable(main main.cc)
+
+# Need manually link all these static libraries:
```

Comment:
When I was working on #29731 I noticed a few places in your iOS tests where you are hard coding the transitive closure of libraries. This is generally an antipattern because it means whenever another dependency is added, we have to find all of the places where we've redundantly specified the transitive closure of libraries and fix them up one-by-one. cmake is supposed to be able to keep track of transitive dependencies, so that you just link against torch, and "It Just Works"; at least, our server-side cmake is working this way. It would be good to figure out why this is not working for the mobile cmake.

Also possibly related: https://github.com/pytorch/pytorch/issues/21737

Reply:
I 100% agree. I thought cmake should be able to track of transitive dependencies, but from what I read it seems during build time we manually track and maintain the dependency with variables like Caffe2_DEPENDENCY_LIBS and link them all into the final shared library (torch.so). And in the public cmake file we add the 2 shared libraries (torch.so, c10.so) manually: https://github.com/pytorch/pytorch/blob/master/cmake/TorchConfig.cmake.in#L56

Shall I try moving these static libraries to "cmake/TorchConfig.cmake.in"?
I know how to test it with mobile build mode but not so sure about non-mobile build that involves other dependent static libraries...

Reply:
> during build time we manually track and maintain the dependency with variables like Caffe2_DEPENDENCY_LIBS

This is true, but that is because we are building a single mega-torch library, and cmake doesn't really understand how to link a bunch of libraries compositionally into one big shared library (I think it's hypothetically possible to build a shared library out of a bunch of static libraries but cmake sure as hell doesn't make it easy). But once you actually make a shared library, and declare that it has public dependencies on other shared libraries, cmake will do the propagation correctly.

TorchConfig.cmake is probably wrong, but I wouldn't try to fix it in this patchset.